ehcache-core/src/main/java/org/ehcache/core/spi/store/heap/SizeOfEngineProvider.java

```diff
@@ -37,4 +37,6 @@ public interface SizeOfEngineProvider extends Service {
    */
 
   SizeOfEngine createSizeOfEngine(ResourceUnit resourceUnit, ServiceConfiguration<?, ?>... serviceConfigs);
```

Comment:
I would have expected this whole class, and `SizeOfEngine` itself to be deprecated.

Reply:
In hindsight, I might have started with marking `SizeOfEngine` deprecated, and then allowed that to propagate through the code base, marking things as deprecated, or suppressing until everything builds cleanly.

Reply:
marked this class `SizeOfEngine` itself as `@Deprecated`,
Now doing the changes as per your 2nd review comment